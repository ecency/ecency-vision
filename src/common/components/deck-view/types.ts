export interface DeckModel {
  data: any;
  dataParams: any[];
  header: {
    title: string;
    icon: JSX.Element | null;
  };
  listItemComponent: any;
  createdAt: Date;
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}
