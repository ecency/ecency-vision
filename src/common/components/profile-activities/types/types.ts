export interface ActivityTypes {
    type: string;
      parent_author?: string;
      parent_permlink?: string;
      author?: string;
      permlink?: string;
      payment?: number;
      payer?: string;
      following?: string;
      community?: string;
      approve?: boolean;
      proposal_ids?: string;
      account?: string;
      id?: string;
      what?: string[];
      voter?: string;
      witness?: string;
      json: string;
    timestamp: string;
  }