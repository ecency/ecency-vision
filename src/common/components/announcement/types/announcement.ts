export interface Announcement {
  id: number;
  title: string;
  description: string;
  button_text: string;
  button_link: string;
}

export interface LaterAnnouncement {
  id: number;
  dateTime: Date;
}
