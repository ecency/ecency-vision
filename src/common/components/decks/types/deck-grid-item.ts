interface BaseDeckGridItem {
  id: string;
  key: number;
  type:
    | "u"
    | "ac"
    | "co"
    | "w"
    | "n"
    | "tr"
    | "to"
    | "s"
    | "cu"
    | "th"
    | "msf"
    | "faq"
    | "wb"
    | "wn";
  settings: Record<string, any>;
}

export interface ReloadableDeckGridItem extends BaseDeckGridItem {
  settings: {
    updateIntervalMs: number;
  };
}

export interface UserDeckGridItem extends ReloadableDeckGridItem {
  type: "u";
  settings: ReloadableDeckGridItem["settings"] & {
    username: string;
    contentType: string;
  };
}

export interface CommunityDeckGridItem extends ReloadableDeckGridItem {
  type: "co";
  settings: UserDeckGridItem["settings"] & {
    tag: string;
  };
}

export interface SearchDeckGridItem extends ReloadableDeckGridItem {
  type: "s";
  settings: ReloadableDeckGridItem["settings"] & {
    query: string;
    author?: string;
    type?: string;
    category?: string;
    tags?: string;
    date?: string;
    sort?: string;
    hideLow?: boolean;
  };
}

export interface WavesDeckGridItem extends ReloadableDeckGridItem {
  type: "th";
  settings: ReloadableDeckGridItem["settings"] & {
    host: string;
  };
}

export type DeckGridItem =
  | BaseDeckGridItem
  | UserDeckGridItem
  | CommunityDeckGridItem
  | SearchDeckGridItem
  | WavesDeckGridItem;
