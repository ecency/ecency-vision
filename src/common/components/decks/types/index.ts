export * from "./deck-grid-item";
export * from "./deck-grid";
export * from "./old-decks";
