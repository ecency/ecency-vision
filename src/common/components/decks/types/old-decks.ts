interface GenericDeck<DF extends Record<string, string | number> | null> {
  createdAt: string;
  dataFilters: DF;
  dataParams: string[];
  header: {
    title: string;
    updateIntervalMs: number;
  };
}

export interface OldNotificationDeck extends GenericDeck<{ type: string; group: string }> {}
export interface OldNonDFDeck extends GenericDeck<null> {}

export type OldDeck = OldNotificationDeck | OldNonDFDeck;
