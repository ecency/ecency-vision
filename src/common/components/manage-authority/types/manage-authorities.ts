import { Account, FullAccount } from "../../../store/accounts/types";
import { PublicKey } from "@hiveio/dhive";

export const Keytype = {
  Owner: "owner",
  Active: "active",
  Posting: "posting",
  Memo: "memo"
};

export interface PublicKeys {
  publicOwnerKey?: string;
  publicActiveKey?: string;
  publicPostingKey?: string;
  publicMemoKey?: string;
}

export interface AccountDataType {
  postingsAuthority: [string, number][];
  posting: [string | PublicKey, number];
  owner: [string | PublicKey, number];
  active: [string | PublicKey, number];
  weight: number;
  memokey: string;
  account: Account;
  publicKeys: {
    publicOwnerKey: string | PublicKey;
    publicActiveKey: string | PublicKey;
    publicPostingKey: string | PublicKey;
    publicMemoKey: string;
  };
}

export const actionType = {
  Revoke: "revoke",
  Keys: "keys",
  Import: "import",
  Reveal: "reveal"
};
