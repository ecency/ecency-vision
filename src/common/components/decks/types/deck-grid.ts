import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  key: string;
  title: string;
  icon?: string;
  columns: DeckGridItem[];
  storageType: "account" | "local";
}

export interface DeckGrids {
  decks: DeckGrid[];
}
