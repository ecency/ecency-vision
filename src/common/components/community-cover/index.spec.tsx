import React from 'react';
import { createBrowserHistory } from 'history';
import renderer from 'react-test-renderer';

import CommunityCover from './index';

import { Theme } from '../../store/global/types';
import { Account } from '../../store/accounts/types';

import {
  globalInstance,
  UiInstance,
  communityInstance1,
  fullAccountInstance
} from '../../helper/test-helper';

jest.mock('../../constants/defaults.json', () => ({
  imageServer: 'https://images.ecency.com'
}));

jest.mock('../../api/hive', () => ({
  getFollowing: () =>
    new Promise((resolve) => {
      resolve([]);
    })
}));

const defProps = {
  history: createBrowserHistory(),
  global: { ...globalInstance },
  community: communityInstance1,
  users: [],
  activeUser: null,
  subscriptions: [],
  ui: UiInstance,
  setActiveUser: () => {},
  updateActiveUser: () => {},
  deleteUser: () => {},
  toggleUIProp: () => {},
  updateSubscriptions: () => {},
  addAccount: () => {}
};

it('(1) Render with loaded account object', () => {
  const account: Account = {
    ...fullAccountInstance,
    name: 'user1',
    profile: {
      cover_image: 'https://img.esteem.app/rwd380.jpg'
    }
  };

  const props = {
    ...defProps,
    account
  };

  const component = renderer.create(<CommunityCover {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('(2) Render with not loaded account object', () => {
  const account: Account = {
    name: 'user1'
  };

  const props = {
    ...defProps,
    account
  };

  const component = renderer.create(<CommunityCover {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('(3) No bg image - Day theme', () => {
  const account: Account = {
    ...fullAccountInstance,
    name: 'user1',
    profile: {}
  };

  const props = {
    ...defProps,
    account
  };

  const component = renderer.create(<CommunityCover {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('(4) No bg image - Night theme', () => {
  const account: Account = {
    ...fullAccountInstance,
    name: 'user1',
    profile: {}
  };

  const props = {
    ...defProps,
    global: { ...globalInstance, ...{ theme: Theme.night } },
    account
  };

  const component = renderer.create(<CommunityCover {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
