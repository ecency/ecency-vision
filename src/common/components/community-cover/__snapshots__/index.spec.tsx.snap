// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Render with loaded account object 1`] = `
<div
  className="community-cover"
>
  <div
    className="cover-image"
    style={
      Object {
        "backgroundImage": "url('https://images.ecency.com/u/hive-148441/cover')",
      }
    }
  />
  <div
    className="community-stats"
  >
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        4,086
      </div>
      <div
        className="stat-label"
      >
        Members
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        $
         
        17,370
      </div>
      <div
        className="stat-label"
      >
        rewards
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        1,796
      </div>
      <div
        className="stat-label"
      >
        posters
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        EN
      </div>
      <div
        className="stat-label"
      >
        language
      </div>
    </div>
  </div>
  <div
    className="controls-holder"
  >
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        Join
      </div>
    </button>
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        New Post
      </div>
    </button>
  </div>
</div>
`;

exports[`(2) Render with not loaded account object 1`] = `
<div
  className="community-cover"
>
  <div
    className="cover-image"
    style={
      Object {
        "backgroundImage": "url('https://images.ecency.com/u/hive-148441/cover')",
      }
    }
  />
  <div
    className="community-stats"
  >
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        4,086
      </div>
      <div
        className="stat-label"
      >
        Members
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        $
         
        17,370
      </div>
      <div
        className="stat-label"
      >
        rewards
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        1,796
      </div>
      <div
        className="stat-label"
      >
        posters
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        EN
      </div>
      <div
        className="stat-label"
      >
        language
      </div>
    </div>
  </div>
  <div
    className="controls-holder"
  >
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        Join
      </div>
    </button>
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        New Post
      </div>
    </button>
  </div>
</div>
`;

exports[`(3) No bg image - Day theme 1`] = `
<div
  className="community-cover"
>
  <div
    className="cover-image"
    style={
      Object {
        "backgroundImage": "url('https://images.ecency.com/u/hive-148441/cover')",
      }
    }
  />
  <div
    className="community-stats"
  >
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        4,086
      </div>
      <div
        className="stat-label"
      >
        Members
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        $
         
        17,370
      </div>
      <div
        className="stat-label"
      >
        rewards
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        1,796
      </div>
      <div
        className="stat-label"
      >
        posters
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        EN
      </div>
      <div
        className="stat-label"
      >
        language
      </div>
    </div>
  </div>
  <div
    className="controls-holder"
  >
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        Join
      </div>
    </button>
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        New Post
      </div>
    </button>
  </div>
</div>
`;

exports[`(4) No bg image - Night theme 1`] = `
<div
  className="community-cover"
>
  <div
    className="cover-image"
    style={
      Object {
        "backgroundImage": "url('https://images.ecency.com/u/hive-148441/cover')",
      }
    }
  />
  <div
    className="community-stats"
  >
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        4,086
      </div>
      <div
        className="stat-label"
      >
        Members
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        $
         
        17,370
      </div>
      <div
        className="stat-label"
      >
        rewards
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        1,796
      </div>
      <div
        className="stat-label"
      >
        posters
      </div>
    </div>
    <div
      className="community-stat"
    >
      <div
        className="stat-value"
      >
        EN
      </div>
      <div
        className="stat-label"
      >
        language
      </div>
    </div>
  </div>
  <div
    className="controls-holder"
  >
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        Join
      </div>
    </button>
    <button
      className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
      onClick={[Function]}
      style={
        Object {
          "outline": "none",
        }
      }
      type="button"
    >
      <div>
        New Post
      </div>
    </button>
  </div>
</div>
`;
